contracts/utils/SignatureValidator.sol

```diff
@@ -1,195 +1,91 @@
-        uint8 v;
-        bytes32 r;
-        bytes32 s;
-        address recovered;
+    // Extract signature type
+    SignatureType signatureType = SignatureType(signatureTypeRaw);
 
+    if (signatureType == SignatureType.Illegal) {
         // Always illegal signature.
         // This is always an implicit option since a signer can create a
         // signature array with invalid type or length. We may as well make
         // it an explicit option. This aids testing and analysis. It is
         // also the initialization value for the enum type.
-        if (signatureType == SignatureType.Illegal) {
-            revert("SignatureValidator#isValidSignature: illegal signature");
-
-            // Signature using EIP712
-        } else if (signatureType == SignatureType.EIP712) {
-            require(_sig.length == 65 || _sig.length == 97, "SignatureValidator#isValidSignature: length 65 or 97 required");
-            r = _sig.readBytes32(0);
-            s = _sig.readBytes32(32);
-            v = uint8(_sig[64]);
-            recovered = ecrecover(_hash, v, r, s);
-            isValid = _signerAddress == recovered;
-            return isValid;
-
-            // Signed using web3.eth_sign() or Ethers wallet.signMessage()
-        } else if (signatureType == SignatureType.EthSign) {
-            require(_sig.length == 65 || _sig.length == 97, "SignatureValidator#isValidSignature: length 65 or 97 required");
-            r = _sig.readBytes32(0);
-            s = _sig.readBytes32(32);
-            v = uint8(_sig[64]);
-            recovered = ecrecover(keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", _hash)), v, r, s);
-            isValid = _signerAddress == recovered;
-            return isValid;
-
-            // Signature verified by wallet contract with data validation.
-        } else if (signatureType == SignatureType.WalletBytes) {
-            isValid = ERC1271_MAGICVALUE == IERC1271Wallet(_signerAddress).isValidSignature(_data, _sig);
-            return isValid;
-
-            // Signature verified by wallet contract without data validation.
-        } else if (signatureType == SignatureType.WalletBytes32) {
-            isValid = ERC1271_MAGICVALUE_BYTES32 == IERC1271Wallet(_signerAddress).isValidSignature(_hash, _sig);
-            return isValid;
-        } else if (signatureType == SignatureType.Wallet) {
-            isValid = isValidWalletSignature(_hash, _signerAddress, _sig);
-            return isValid;
-        }
 
-        // Anything else is illegal (We do not return false because
-        // the signature may actually be valid, just not in a format
-        // that we currently support. In this case returning false
-        // may lead the caller to incorrectly believe that the
-        // signature was invalid.)
-        revert("SignatureValidator#isValidSignature: unsupported signature");
+        revert("SignatureValidator#isValidSignature: illegal signature");
```

Comment:
Maybe we can skip or replace `SignatureValidator#isValidSignature`?

Reply:
@charlesjhongc Should the prefix be updated?

Reply:
Updated in 472d85b0f54876e716461f8633e9cbcc3da1e16e